chore: support lazy materialization for more queries. (#11738)
datafuselabs/databend: docs/doc/13-sql-reference/20-system-tables/system-settings.md

| group_by_two_level_threshold          | 20000        | 20000        | SESSION | Sets the number of keys in a GROUP BY operation that will trigger a two-level aggregation.                                                                                          | UInt64 |
| hide_options_in_show_create_table     | 1            | 1            | SESSION | Hides table-relevant information, such as SNAPSHOT_LOCATION and STORAGE_FORMAT, at the end of the result of SHOW TABLE CREATE.                                                      | UInt64 |
| input_read_buffer_size                | 1048576      | 1048576      | SESSION | Sets the memory size in bytes allocated to the buffer used by the buffered reader to read data from storage.                                                                        | UInt64 |

> | lazy_read_threshold                   | 1000         | 1000         | SESSION | Enable lazy materialization and set the limit threshold of queries with limit. Set the value to 0 to disable this setting.                                                               | UInt64 |

Reviewer: ```suggestion
| lazy_read_threshold                   | 1000         | 1000         | SESSION | Sets the maximum LIMIT in a query to enable lazy read optimization. Setting it to 0 disables the optimization.                                                               | UInt64 |
```
